### ISSUE TITLE
Optimise runtime speed

### ISSUE THREAD

**Issue by AoifeHughes** [PandemiaProject/pandemia#63]

**Comment by AoifeHughes** [comment]
Used to run faster, we made changes which slowed things down! I'll get onto this!

@James-Thompson-724 do you have a rough time estimate for what your run time was before and what it is now, just for my own references for checking this moving forward!

**Comment by James-Thompson-724** [comment]
See this issue: https://github.com/PandemiaProject/pandemia/issues/59

**Comment by James-Thompson-724** [comment]
Running without parallelism, on a single CPU, I'm getting:

`heterogeneous_baseline.yaml` (64bit)
1.57s/it
`heterogeneous_baseline.yaml` (32bit)
1.26s/it
`homogeneous_baseline.yaml` (64bit)
0.79s/it
`homogeneous_baseline.yaml` (32bit)
0.60s/it

where I've swapped the data types from 64bit to 32bit to see if there's a difference (except for the prng state, which has to be 64bit). So, in fact, using 32bit data types, we're back to the speed of the simulator in `main`. So to recover that performance, we simply need to update the data types appropriately.

**Comment by AoifeHughes** [comment]
@James-Thompson-724 did you change the data types in C and the numpy arrays to get this? I was about start experimenting with some similar stuff but if you have a branch testing this, it would be a good starting point.

**Comment by James-Thompson-724** [comment]
@AoifeHughes Yeah that's what I did, but for some reason I didn't save the changes anywhere(!). The variables inside the C functions `rotl` and `next` need to be 64bit, and the random state `uint64_t * s` / `uint64_t * random_state` needs to be 64bit. Also, the numpy array `infections` appearing in `simulator.py` needs to be 64bit, since those values can sum to more than the 32bit limit, as do the arrays `vector_region.random_state`. So, what I did was isolate those objects, and find and replace the data types on everything else (apologies for the slow reply, just noticed your comment!)

**Comment by James-Thompson-724** [comment]
All 64 bit integers have been converted to 32 bit, returning runtimes to the previous benchmark. In particular, a new prng, using 32 bit integers, has been implemented.
